File: CSMA-CA/BaseStation.java
public class BaseStation {

    public MobileStation curSender; //keeps track of the mobile station ur working with
    private boolean Busy;

    State state = State.IDLE;
    public boolean isBusy() {
        return Busy;
    }

    public void setBusy(boolean busy) {
        Busy = busy;
    }

    public void activeAction()
    {
        if(this.state == State.IDLE)
        {
            return;
        }
        else if (this.state == State.SIFS_before_emitCTS)
        {
            // delay SIFS_before_emitCTS: 3
            this.state = State.emitCTS;
        }
        else if(this.state == State.emitCTS)
        {
            // delay SIFS_before_emitCTS: 15
            // send your CTS using reception function
            this.state = State.SIFS_before_rcvPKT;
        }
        else if(this.state == State.SIFS_before_rcvPKT)
        {
            // delay SIFS_before_rcvPKT: 3
            this.state = State.rcvPKT;
        }
        else if(this.state == State.rcvPKT)
        {
            // delay rcvPKT: 61
            this.state = State.IDLE;
        }
        else if(this.state == State.SIFS_before_emitACK)
        {
            // delay SIFS_before_emitACK : 3
            this.state = State.emitACK;
        }
        else if(this.state == State.emitACK)
        {
            // send your ACK using reception function
            this.state = State.IDLE;
        }

    }
    public void receptionAction(Packet packet)
    {
        if(this.state == State.IDLE)
        {
            if(!packet.isCorrupted() && packet.getType() == Packet.RTS)
            {
                this.curSender = packet.getMobileStation();
                this.state = State.SIFS_before_emitCTS;
            }
        }
        else if (this.state == State.rcvPKT)
        {
            if(!packet.isCorrupted() && packet.getType() == 3 &&  packet.getMobileStation() == this.curSender) // 3 for PKT
            {
                this.curSender = packet.getMobileStation();
                this.state = State.SIFS_before_emitACK;
            }
        }
    }
}

File: baseStation.py
from state import State
from mobileStation import MobileStation
from packet import Packet

class BaseStation():

    # still need to implement wait delay

    def __init__(self):
        self.isBusy = False
        self.state = State(State.IDLE)
        self.currSender = None


    def activeAction(self):
        if self.state == State.IDLE:
            return
        elif self.state == State.SIFS_before_emitCTS:
            # delay SIFS_before_emitCTS: 3
            self.state = State.emitCTS

        elif self.state == State.emitCTS:
            # delay SIFS_before_emitCTS: 15
            # send your CTS using reception function
            self.state = State.SIFS_before_rcvPKT

        elif self.state == State.SIFS_before_rcvPKT:
            # delay SIFS_before_rcvPKT: 3
            self.state = State.rcvPKT

        elif self.state == State.rcvPKT:
            # delay rcvPKT: 61
            self.state = State.IDLE

        elif self.state == State.SIFS_before_emitACK:
            # delay SIFS_before_emitACK : 3
            self.state = State.emitACK

        elif self.state == State.emitACK:
            # send your ACK using reception function
            self.state = State.IDLE


    def receptionAction(self, packet):
        if (self.state == State.IDLE) and (not packet.corrupted) and (packet.type == Packet.RTS):
            self.currSender = packet.mobileStation
            self.state = State.SIFS_before_emitACK
        elif (self.state == State.rcvPKT) and (not packet.corrupted):
            if (packet.type == Packet.PKT) and (packet.mobileStation == self.currSender):
                self.state = State.SIFS_before_emitACK
            else:
                self.state.changeTo(State.IDLE)